// evergreen-ui/esm/themes/classic/components/checkbox.js
import { getPrimaryButtonStylesForIntent } from '../deprecated/helpers';
var baseStyle = {};
var appearances = {
  "default": {
    _base: {
      color: 'transparent',
      background: function background(theme) {
        return "linear-gradient(to top, ".concat(theme.scales.neutral.N2A, ", white)");
      },
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N3A);
      }
    },
    _disabled: {
      cursor: 'not-allowed',
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A);
      },
      background: 'scales.neutral.N2A'
    },
    _hover: {
      background: function background(theme) {
        return "linear-gradient(to top, ".concat(theme.scales.neutral.N2A, ", ").concat(theme.scales.neutral.N1A, ")");
      },
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
      }
    },
    _focus: {
      boxShadow: function boxShadow(theme) {
        return "0 0 0 2px ".concat(theme.scales.blue.B4A, ", inset 0 0 0 1px ").concat(theme.scales.neutral.N5A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N3A);
      }
    },
    _active: {
      background: 'scales.blue.B3A',
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.blue.B5A);
      }
    },
    _checked: {
      color: 'white',
      background: getPrimaryButtonStylesForIntent().linearGradient.base,
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N5A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
      }
    },
    _checkedHover: {
      color: 'white',
      background: getPrimaryButtonStylesForIntent().linearGradient.hover,
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N5A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
      }
    },
    _checkedDisabled: {
      color: 'scales.neutral.N6A',
      background: function background(theme) {
        return "linear-gradient(to top, ".concat(theme.scales.neutral.N2A, ", ").concat(theme.scales.neutral.N1A, ")");
      },
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
      }
    },
    _checkedActive: {
      color: 'white',
      background: getPrimaryButtonStylesForIntent().linearGradient.active,
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A, ", inset 0 -1px 1px 0 ").concat(theme.scales.neutral.N2A);
      }
    }
  }
};
var sizes = {};
export default {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
