// evergreen-ui/commonjs/themes/default/tokens/fills.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = _interopRequireDefault(require("./colors"));

/**
 * Used for matching background colors and foreground colors
 */
var fills = {
  neutral: {
    color: _colors["default"].gray800,
    backgroundColor: _colors["default"].gray200
  },
  blue: {
    color: _colors["default"].blue600,
    backgroundColor: _colors["default"].blue100
  },
  red: {
    color: _colors["default"].red700,
    backgroundColor: _colors["default"].red100
  },
  orange: {
    color: _colors["default"].orange700,
    backgroundColor: _colors["default"].orange100
  },
  yellow: {
    color: _colors["default"].yellow800,
    backgroundColor: _colors["default"].yellow100
  },
  green: {
    color: _colors["default"].green700,
    backgroundColor: _colors["default"].green100
  },
  teal: {
    color: _colors["default"].teal800,
    backgroundColor: _colors["default"].teal100
  },
  purple: {
    color: _colors["default"].purple600,
    backgroundColor: _colors["default"].purple100
  }
};
var _default = fills;
exports["default"] = _default;
