// evergreen-ui/commonjs/themes/default/tokens/intents.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _colors = _interopRequireDefault(require("./colors"));

var intents = {
  info: {
    background: _colors["default"].blue25,
    border: _colors["default"].blue500,
    text: _colors["default"].blue600,
    icon: _colors["default"].blue500
  },
  success: {
    background: _colors["default"].green25,
    border: _colors["default"].green500,
    text: _colors["default"].green700,
    icon: _colors["default"].green500
  },
  warning: {
    background: _colors["default"].orange25,
    border: _colors["default"].orange500,
    text: _colors["default"].orange700,
    icon: _colors["default"].orange500
  },
  danger: {
    background: _colors["default"].red25,
    border: _colors["default"].red500,
    text: _colors["default"].red600,
    icon: _colors["default"].red500
  }
};
var _default = intents;
exports["default"] = _default;
