// evergreen-ui/commonjs/themes/classic/components/link.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _tinycolor = _interopRequireDefault(require("tinycolor2"));

var baseStyle = {
  color: function color(_, _ref) {
    var _color = _ref.color;
    return "colors.".concat(getThemeKeyForColor(_color), ".base");
  },
  _hover: {
    color: function color(_, _ref2) {
      var _color2 = _ref2.color;
      return (0, _tinycolor["default"])("colors.".concat(getThemeKeyForColor(_color2), ".base")).lighten(10).toString();
    }
  },
  _active: {
    color: function color(theme, _ref3) {
      var _color3 = _ref3.color;
      return (0, _tinycolor["default"])(theme.colors[getThemeKeyForColor(_color3)].base).darken(10).toString();
    }
  },
  _focus: {
    boxShadow: function boxShadow(theme, _ref4) {
      var color = _ref4.color;
      return "0 0 0 2px ".concat((0, _tinycolor["default"])(theme.colors[getThemeKeyForColor(color)].base).setAlpha(0.4).toString());
    }
  }
};

var getThemeKeyForColor = function getThemeKeyForColor(color) {
  switch (color) {
    case 'blue':
    case 'neutral':
    case 'green':
      return color;

    default:
      return 'blue';
  }
};

var appearances = {};
var sizes = {};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
