// evergreen-ui/commonjs/themes/classic/components/tag-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var baseStyle = {
  paddingY: '2px',
  paddingX: '4px',
  backgroundColor: 'white',
  borderRadius: 'radii.1'
};
var appearances = {
  "default": {
    backgroundColor: 'white',
    boxShadow: function boxShadow(theme) {
      return "inset 0 0 0 1px ".concat(theme.scales.neutral.N5A, ", inset 0 1px 2px ").concat(theme.scales.neutral.N4A);
    },
    _focused: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 2px ".concat(theme.scales.neutral.N4A, ", inset 0 0 0 1px ").concat(theme.scales.blue.B7, ", ").concat(theme.shadows.focusRing);
      },
      zIndex: 'zIndices.focused'
    },
    _invalid: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.palette.red.base, ", inset 0 1px 2px ").concat(theme.scales.neutral.N4A);
      }
    },
    _disabled: {
      boxShadow: function boxShadow(theme) {
        return "inset 0 0 0 1px ".concat(theme.scales.neutral.N4A);
      },
      backgroundColor: 'scales.neutral.N2'
    }
  }
};
var sizes = {};
var _default = {
  baseStyle: baseStyle,
  appearances: appearances,
  sizes: sizes
};
exports["default"] = _default;
