// evergreen-ui/commonjs/file-uploader/src/FileUploader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _lodash = _interopRequireDefault(require("lodash.isempty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _uiBox = _interopRequireDefault(require("ui-box"));

var _constants = require("../../constants");

var _formField = require("../../form-field");

var _hooks = require("../../hooks");

var _icons = require("../../icons");

var _arrayToCsv = _interopRequireDefault(require("../../lib/array-to-csv"));

var _isFunction = _interopRequireDefault(require("../../lib/is-function"));

var _safeInvoke = _interopRequireDefault(require("../../lib/safe-invoke"));

var _scales = require("../../scales");

var _theme = require("../../theme");

var _typography = require("../../typography");

var _FileCard = _interopRequireDefault(require("./FileCard"));

var _getFileDataTransferItems = _interopRequireDefault(require("./utils/get-file-data-transfer-items"));

var _messages = require("./utils/messages");

var _splitFiles2 = _interopRequireDefault(require("./utils/split-files"));

var _excluded = ["acceptedMimeTypes", "description", "disabled", "hint", "isRequired", "label", "labelFor", "maxFiles", "maxSizeInBytes", "onAccepted", "onChange", "onRejected", "onRemove", "renderFile", "validationMessage", "values"],
    _excluded2 = ["className"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var UploaderState = {
  Initial: 'initial',
  Dragging: 'dragging',
  Error: 'error'
};
var disabledPseudoSelector = "&[aria-disabled='true']";
var dragHoverPseudoSelector = "&[data-state='".concat(UploaderState.Dragging, "']");
var invalidPseudoSelector = "&[aria-invalid='true']";
var hoverPseudoSelector = "&:hover:not(".concat(disabledPseudoSelector, "):not(").concat(dragHoverPseudoSelector, "):not(").concat(invalidPseudoSelector, ")");
var styleModifiers = {};
var pseudoSelectors = {
  _focus: '&:focus',
  _hover: hoverPseudoSelector,
  _hoverBrowseCopy: "".concat(hoverPseudoSelector, " span:first-of-type"),
  _hoverOrDragCopy: "".concat(hoverPseudoSelector, " span:last-of-type"),
  _dragHover: dragHoverPseudoSelector,
  _disabled: disabledPseudoSelector,
  _invalid: invalidPseudoSelector
};
var internalStyles = {};
var FileUploader = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var acceptedMimeTypes = props.acceptedMimeTypes,
      description = props.description,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      hint = props.hint,
      isRequired = props.isRequired,
      label = props.label,
      labelFor = props.labelFor,
      maxFiles = props.maxFiles,
      maxSizeInBytes = props.maxSizeInBytes,
      onAccepted = props.onAccepted,
      onChange = props.onChange,
      onRejected = props.onRejected,
      onRemove = props.onRemove,
      renderFile = props.renderFile,
      validationMessageProp = props.validationMessage,
      values = props.values,
      rest = (0, _objectWithoutProperties2["default"])(props, _excluded);

  var _useTheme = (0, _theme.useTheme)(),
      colors = _useTheme.colors;

  var _useStyleConfig = (0, _hooks.useStyleConfig)('FileUploader', styleModifiers, pseudoSelectors, internalStyles),
      className = _useStyleConfig.className,
      boxProps = (0, _objectWithoutProperties2["default"])(_useStyleConfig, _excluded2);

  var _useState = (0, _react.useState)(UploaderState.Initial),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      state = _useState2[0],
      setState = _useState2[1];

  var _useState3 = (0, _react.useState)(''),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      validationMessage = _useState4[0],
      setValidationMessage = _useState4[1];
  /**
   * The underlying <input type="file" /> DOM element won't accept the same file after it has been
   * picked unless it is rerendered manually - if a user selects and removes a file, they should
   * still be able to pick it again without refreshing the page.
   * https://stackoverflow.com/a/45846251
   */


  var _useState5 = (0, _react.useState)(0),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      fileInputKey = _useState6[0],
      setFileInputKey = _useState6[1];

  var fileInputRef = (0, _react.useRef)(null);
  var orDragCopy = "or drag ".concat(maxFiles === 1 ? 'a file' : 'files', " here"); // If the dropzone is meant to be a single file input and we already have a file, don't render
  // the dropzone which will always result in rejected files/errors.

  var renderDropzone = maxFiles !== 1 || (0, _lodash["default"])(values);
  var resetState = (0, _react.useCallback)(function () {
    setState(UploaderState.Initial);
    setValidationMessage('');
  }, []);
  var handleChange = (0, _react.useCallback)(
  /**
   * @param {FileList} fileList
   */
  function (fileList) {
    setFileInputKey(function (prev) {
      return prev + 1;
    });

    if ((0, _lodash["default"])(fileList)) {
      (0, _safeInvoke["default"])(onChange, []);
      return;
    }

    var files = (0, _toConsumableArray2["default"])(fileList);
    (0, _safeInvoke["default"])(onChange, files);

    var _splitFiles = (0, _splitFiles2["default"])(files, {
      maxSizeInBytes: maxSizeInBytes,
      acceptedMimeTypes: acceptedMimeTypes,
      currentFileCount: values === null || values === void 0 ? void 0 : values.length,
      maxFiles: maxFiles
    }),
        accepted = _splitFiles.accepted,
        rejected = _splitFiles.rejected;

    if (!(0, _lodash["default"])(accepted)) {
      (0, _safeInvoke["default"])(onAccepted, accepted);
    }

    if (!(0, _lodash["default"])(rejected)) {
      (0, _safeInvoke["default"])(onRejected, rejected);
    }
  }, [acceptedMimeTypes, maxFiles, maxSizeInBytes, onAccepted, onChange, onRejected, values === null || values === void 0 ? void 0 : values.length]);
  var handleClick = (0, _react.useCallback)(function () {
    if (disabled) {
      return;
    }

    if (fileInputRef.current == null) {
      return;
    }

    fileInputRef.current.click();
  }, [disabled]);
  var handleDragOver = (0, _react.useCallback)(
  /**
   * @param {React.DragEvent<HTMLDivElement>} event
   */
  function (event) {
    event.preventDefault();
    event.stopPropagation();
    event.dataTransfer.dropEffect = 'copy';

    if (disabled) {
      return;
    }

    var dragItems = (0, _getFileDataTransferItems["default"])(event.dataTransfer.items);
    var draggingCount = dragItems.length;

    var _ref = values !== null && values !== void 0 ? values : [],
        currentCount = _ref.length;

    if (maxFiles == null || maxFiles < 0) {
      setState(UploaderState.Dragging);
      return;
    }

    if (draggingCount > maxFiles || draggingCount + currentCount > maxFiles) {
      setValidationMessage((0, _messages.getMaxFilesMessage)(maxFiles));
      setState(UploaderState.Error);
      return;
    }

    setState(UploaderState.Dragging);
  }, [disabled, maxFiles, values]);
  var handleDragLeave = (0, _react.useCallback)(function () {
    return resetState();
  }, [resetState]);
  var handleDrop = (0, _react.useCallback)(
  /**
   * @param {React.DragEvent<HTMLDivElement>} event
   */
  function (event) {
    event.preventDefault();
    event.stopPropagation();

    if (disabled) {
      return;
    }

    resetState();
    handleChange(event.dataTransfer.files);
  }, [disabled, handleChange, resetState]);
  var handleInputChange = (0, _react.useCallback)(
  /**
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  function (event) {
    // Theoretically the input should not be accessible at all when disabled,
    // but this should act as a safeguard
    if (disabled) {
      return;
    }

    handleChange(event.target.files);
  }, [disabled, handleChange]);
  var handleKeyDown = (0, _react.useCallback)(
  /**
   * @param {React.KeyboardEvent<HTMLDivElement>} event
   */
  function (event) {
    if (event.key !== _constants.Key.Enter && event.key !== _constants.Key.Space) {
      return;
    }

    event.preventDefault();
    handleClick();
  }, [handleClick]);
  return /*#__PURE__*/_react["default"].createElement(_uiBox["default"], {
    ref: ref
  }, /*#__PURE__*/_react["default"].createElement(_formField.FormField, {
    label: label,
    labelFor: labelFor,
    description: description,
    hint: hint,
    isRequired: isRequired // Always override the validationMessage from prop if we have a message to display from dragging
    ,
    validationMessage: !(0, _lodash["default"])(validationMessage) ? /*#__PURE__*/_react["default"].createElement(_typography.Text, {
      color: colors.red500
    }, validationMessage) : validationMessageProp
  }, renderDropzone && /*#__PURE__*/_react["default"].createElement(_uiBox["default"], (0, _extends2["default"])({
    "aria-disabled": disabled,
    "aria-invalid": state === UploaderState.Error,
    className: className,
    "data-state": state,
    onClick: handleClick,
    onDragLeave: handleDragLeave,
    onDragOver: handleDragOver,
    onDrop: handleDrop,
    onKeyDown: handleKeyDown,
    tabIndex: disabled ? undefined : 0
  }, boxProps, rest), /*#__PURE__*/_react["default"].createElement(_uiBox["default"], {
    accept: (0, _arrayToCsv["default"])(acceptedMimeTypes),
    display: "none",
    is: "input",
    tabIndex: -1,
    key: fileInputKey,
    multiple: maxFiles !== 1,
    onChange: handleInputChange,
    ref: fileInputRef,
    type: "file"
  }), /*#__PURE__*/_react["default"].createElement(_uiBox["default"], {
    alignItems: "center",
    backgroundColor: disabled ? colors.gray90 : colors.gray200,
    borderRadius: "50%",
    display: "flex",
    height: (0, _scales.majorScale)(7),
    justifyContent: "center",
    pointerEvents: "none",
    width: (0, _scales.majorScale)(7)
  }, /*#__PURE__*/_react["default"].createElement(_icons.UploadIcon, {
    color: disabled ? colors.gray400 : colors.gray500,
    size: (0, _scales.majorScale)(3)
  })), /*#__PURE__*/_react["default"].createElement(_typography.Paragraph, {
    marginTop: (0, _scales.majorScale)(3),
    pointerEvents: "none"
  }, /*#__PURE__*/_react["default"].createElement(_typography.Text, {
    color: disabled ? colors.gray500 : colors.blue400
  }, "Browse "), /*#__PURE__*/_react["default"].createElement(_typography.Text, {
    color: disabled ? colors.gray500 : colors.gray700
  }, orDragCopy)))), /*#__PURE__*/_react["default"].createElement(_uiBox["default"], {
    marginTop: (0, _scales.majorScale)(2)
  }, values === null || values === void 0 ? void 0 : values.map(
  /**
   * @param {File} file
   * @param {number} index
   */
  function (file, index) {
    return (0, _isFunction["default"])(renderFile) ? renderFile(file, index) : /*#__PURE__*/_react["default"].createElement(_FileCard["default"], {
      key: "".concat(file.name, "-").concat(index),
      name: file.name,
      onRemove: (0, _isFunction["default"])(onRemove) ? function () {
        return onRemove(file);
      } : undefined,
      sizeInBytes: file.size,
      type: file.type
    });
  })));
}));
FileUploader.propTypes = _objectSpread(_objectSpread({}, _formField.FormField.propTypes), {}, {
  /**
   * MIME types (not file extensions) to accept
   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
   */
  acceptedMimeTypes: _propTypes["default"].array,

  /**
   * When true, displays a disabled state where drops don't fire and the native browser picker doesn't open
   */
  disabled: _propTypes["default"].bool,

  /**
   * Maximum number of files to accept
   */
  maxFiles: _propTypes["default"].number,

  /**
   * Maximum size of an **individual** file to accept
   */
  maxSizeInBytes: _propTypes["default"].number,

  /**
   * Callback for when files are accepted via drop or the native browser picker
   * @type {(files: File[]) => void}
   */
  onAccepted: _propTypes["default"].func,

  /**
   * Callback for when files are added via drop or the native browser picker, which includes both
   * the accepted and rejected files
   * @type {(files: File[]) => void}
   */
  onChange: _propTypes["default"].func,

  /**
   * Callback for when files are rejected via drop or the native browser picker
   * @type {(fileRejections: FileRejection[]) => void}
   */
  onRejected: _propTypes["default"].func,

  /**
   * Callback to fire when a file should be removed
   * @type {(file: File) => void}
   */
  onRemove: _propTypes["default"].func,

  /**
   * Custom render function for displaying the file underneath the uploader
   * @type {(file: File, index: number) => React.ReactNode}
   */
  renderFile: _propTypes["default"].func,

  /**
   * File values to render underneath the uploader
   * @type {File}
   */
  values: _propTypes["default"].array
});
var _default = FileUploader;
exports["default"] = _default;
